lib/kms.js

```diff
@@ -1,70 +1,54 @@
 /*!
- * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ * Copyright (c) 2020-2021 Digital Bazaar, Inc. All rights reserved.
  */
 import * as bedrock from 'bedrock';
 import * as brHttpsAgent from 'bedrock-https-agent';
-import {
-  AsymmetricKey,
-  Hmac,
-  Kek,
-  KeystoreAgent,
-  KeyAgreementKey,
-  KmsClient
-} from 'webkms-client';
+import {KeystoreAgent, KmsClient} from '@digitalbazaar/webkms-client';
 
-export async function createKeystore({capabilityAgent, referenceId} = {}) {
-  const {kmsBaseUrl} = bedrock.config.tokenizer;
+export async function createKeystore({
+  controller, kmsModule, meterId, meterCapabilityInvocationSigner,
+  referenceId, applyIpAllowList = true
+} = {}) {
+  // use default KMS module if not provided
+  if(!kmsModule) {
+    kmsModule = bedrock.config.tokenizer.kms.defaultKmsModule;
+  }
+  // use configured meter ID if not provided
+  if(!meterId) {
+    meterId = bedrock.config.tokenizer.kms.meterId;
+  }
   // create keystore
-  const config = {
+  const keystoreConfig = {
     sequence: 0,
-    controller: capabilityAgent.id
+    controller,
+    kmsModule,
+    meterId
   };
+  if(applyIpAllowList) {
+    keystoreConfig.ipAllowList = bedrock.config.tokenizer.kms.ipAllowList;
+  }
   if(referenceId) {
-    config.referenceId = referenceId;
+    keystoreConfig.referenceId = referenceId;
   }
   const {httpsAgent} = brHttpsAgent;
   return KmsClient.createKeystore({
-    url: `${kmsBaseUrl}/keystores`,
-    config,
+    url: `${bedrock.config.tokenizer.kms.baseUrl}/keystores`,
+    config: keystoreConfig,
+    invocationSigner: meterCapabilityInvocationSigner,
```

Comment:
This doesn't seem right, the invocation signer is for signing the zcap invocation for creating the keystore, it isn't for signing the meter zcap invocation (the KMS system will be signing that).

Reply:
Can change the name to`invocationSigner`.